src/ClientData/ScopeIdProvider.cpp

```diff
@@ -81,10 +82,16 @@ std::optional<ScopeId> NameEqualityScopeIdProvider::ProvideId(const TimerInfo& t
 
   const ScopeInfo scope_info{timer_info.api_scope_name(), scope_type};
 
-  const auto it = scope_info_to_id_.find(scope_info);
-  if (it != scope_info_to_id_.end()) {
-    return it->second;
+  {
+    absl::ReaderMutexLock reader_lock{&mutex_};
+    const auto it = scope_info_to_id_.find(scope_info);
+    if (it != scope_info_to_id_.end()) {
+      return it->second;
+    }
```

Comment:
I think you need to do this check again the writer lock because the id could have been added between those two locks. (There is no way to convert a read lock into a write lock if you were wondering)